crates/modelardb_compression/src/types.rs

```diff
@@ -760,28 +736,72 @@ mod tests {
         expected_segment_max_value: Value,
         expected_segment_values_length: usize,
     ) {
+        let (
+            model_start_index,
+            model_end_index,
+            segment_min_value,
+            segment_max_value,
+            segment_values,
+        ) = create_segment_assert_segment_and_return_model_values(
+            &uncompressed_values,
+            SWING_ID,
+            expected_model_end_index,
+            expected_model_min_value,
+            expected_model_max_value,
+            expected_model_values_length,
+            expected_segment_min_value,
+            expected_segment_max_value,
+            expected_segment_values_length,
+        );
+
+        let model_first_value = uncompressed_values.value(model_start_index);
+        let model_last_value = uncompressed_values.value(model_end_index);
+
+        let (segment_first_value, segment_last_value) =
+            CompressedSegmentBuilder::decode_values_for_swing(
+                segment_min_value,
+                segment_max_value,
+                &segment_values,
+            );
+
+        assert_eq!(model_first_value, segment_first_value);
+        assert_eq!(model_last_value, segment_last_value);
+    }
+
+    fn create_segment_assert_segment_and_return_model_values(
```

Comment:
Seems odd to mention the return value in the function name like this, maybe rename to something like `create_and_assert_expected_segment`.

Reply:
Fixed.